kernel/aster-nix/src/sched/priority_scheduler.rs

```diff
@@ -36,9 +36,22 @@ impl<T: PreemptSchedInfo> PreemptScheduler<T> {
     }
 
     /// Selects a cpu for task to run on.
-    fn select_cpu(&self, _runnable: &Arc<T>) -> u32 {
-        // FIXME: adopt more reasonable policy once we fully enable SMP.
-        0
+    fn select_cpu(&self, runnable: &Arc<T>) -> u32 {
```

Comment:
Add the following code to make sure that tasks do not migrate one CPU core to another on every enqueue operation. Frequent migration will lead to terrible performance due to CPU cache miss.

```rust
    // If the CPU of a runnable task has been set before, keep scheduling the task to the CPU core.
    // TODO: migrate tasks at proper timings for load balancing between CPU cores.
    if let Some(cpu_id) = runnable.get() {
        return cpu_id;
    }
```

Reply:
Currently the task sticks to that CPU once selected. As `select_cpu` is an internal API that is only called when first enqueuing a task, I think there's no need to check former CPU ID. We can mark this function as first selection as this PR is not intended to develop CPU migration.

Reply:
> As select_cpu is an internal API that is only called when first enqueuing a task

This is not true. The `enqueue` method is called whenever a task becomes runnable. This happens when a new task is spawn, but also when when a sleeping task is woken. So the method should not be named `first_select_cpu`.